main.tf

```diff
@@ -282,7 +292,7 @@ resource "azurerm_key_vault" "service" {
 resource "azurerm_storage_account" "service" {
   for_each = toset(var.regions)
 
-  name                     = "${local.service_name}${each.key}${local.environment_differentiator_short}${var.environment}"
+  name                     = "${local.service_name}${try(local.region_map[each.key], each.key)}${local.environment_differentiator_short}${var.environment}"
```

Comment:
Any particular reason why you went with:

`try(local.region_map[each.key], each.key)`

Instead of:

`${lookup(local.region_map, each.key)}`

I know the try adds some safety, but since we control the creation of the map we should trust it. The safety could end up masking an issue that I would rather have error so we can fix it.

Reply:
Because lookup only returns default if you use the full form:
```
lookup(map, key, not_found_val)
```

And I hadn't realized you could just directly access keys in a map with the bracket operators. To me, it was clearer that we were "trying to get a value" and if it fails, use the alternate value (each.key in this case)

Reply:
You are right about the full form and I think it makes sense to use that since it should be more clear that the third value is a fallback. So can we change these to:

`${lookup(local.region_map, each.key, each.key)}`

I am just not a big fan that the try will also fallback for improper format, etc.

Reply:
Incorporated.